src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/BoxingInstructions.cs

```diff
@@ -172,77 +144,90 @@ public static uint[] GetBoxingTokens(MethodInfo method)
                 MethodDefinitionHandle methodDefHandle = (MethodDefinitionHandle)MetadataTokens.Handle(method.MetadataToken);
                 MethodDefinition methodDef = mdReader.GetMethodDefinition(methodDefHandle);
 
-                MethodSignature<uint> methodSignature = methodDef.DecodeSignature(new BoxingTokensSignatureProvider(), genericContext: null);
+                MethodSignature<uint?> methodSignature = methodDef.DecodeSignature(new BoxingTokensSignatureProvider(), genericContext: null);
```

Comment:
If you change `BoxingTokenSignatureProvider` to return `SpecialCaseBoxingTypes.Unknown` instead of returning `null` for unsupported types, then this code path can be reverted and no nullability checks are required.

Reply:
Changing the provider to return `ParameterBoxingInstructions` rather than `uint` would further simplify this.